src/main/java/com/google/jenkins/plugins/k8sengine/KubeConfig.java

```diff
@@ -35,38 +36,41 @@
   private static final String API_VERSION = "v1";
   private static final String CONFIG_KIND = "Config";
 
-  private ImmutableList<Object> contexts;
-  private ImmutableList<Object> clusters;
-  private ImmutableList<Object> users;
+  private List<Object> contexts;
+  private List<Object> clusters;
+  private List<Object> users;
   private String currentContext;
 
   private KubeConfig() {}
 
   /** @return This config's contexts. */
-  public ImmutableList<Object> getContexts() {
+  public List<Object> getContexts() {
     return contexts;
   }
 
-  private void setContexts(ImmutableList<Object> contexts) {
-    this.contexts = Preconditions.checkNotNull(contexts);
+  private void setContexts(List<Object> contexts) {
+    Objects.requireNonNull(contexts);
+    this.contexts = contexts;
   }
 
   /** @return This config's clusters. */
-  public ImmutableList<Object> getClusters() {
+  public List<Object> getClusters() {
     return clusters;
   }
 
-  private void setClusters(ImmutableList<Object> clusters) {
-    this.clusters = Preconditions.checkNotNull(clusters);
+  private void setClusters(List<Object> clusters) {
+    Objects.requireNonNull(clusters);
+    this.clusters = clusters;
   }
 
   /** @return This config's users. */
-  public ImmutableList<Object> getUsers() {
+  public List<Object> getUsers() {
```

Comment:
@jtnord @alecharp  this break backward compat. I agree it;s a bit agressive I can revert (but it hurts my eyes)